File: src/Moryx.Runtime.Endpoints/Databases/Models/ModelConfiguratorModel.cs
// Copyright (c) 2026 Phoenix Contact GmbH & Co. KG
// Licensed under the Apache License, Version 2.0

using Moryx.Serialization;

namespace Moryx.Runtime.Endpoints.Databases.Models
{
    /// <summary>
    /// Database configurator options model
    /// </summary>
    public class ModelConfiguratorModel
    {
        /// <summary>
        /// Name of the configurator
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// Assembly qualified name of the configurator
        /// </summary>
        public string ConfiguratorType { get; set; }

        /// <summary>
        /// Keys for the connection string to map entries
        /// </summary>
        public IReadOnlyDictionary<string, string> ConnectionStringKeys { get; set; }

        /// <summary>
        /// Prototype of the connection string
        /// </summary>
        public string ConnectionStringPrototype { get; set; }

        /// <summary>
        /// Properties to be configured for the given configurator
        /// </summary>
        public Entry ConfigPrototype { get; set; }
    }
}

File: src/Moryx.CommandCenter.Web/src/databases/models/ModelConfiguratorModel.ts
/*
 * Copyright (c) 2026, Phoenix Contact GmbH & Co. KG
 * Licensed under the Apache License, Version 2.0
*/

import Entry from "../../modules/models/Entry";

export class ModelConfiguratorModel {
    public name: string;
    public configuratorType: string;
    public connectionStringKeys: Record<string, string>;
    public configPrototype: Entry;
    public connectionStringPrototype: string;
}
